File: algos3/Algos3/Task1.cs
namespace Algos3;

public static class Task1
{
    public static void SelectionSortStep(int[] array, int i)
    {
        if (i >= array.Length - 1)
            return;

        var minIndex = i;

        for (int j = i + 1; j < array.Length; j++)
        {
            if (array[minIndex] > array[j])
                minIndex = j;
        }

        Swap(minIndex, i, array);
    }

    public static void Swap(int x, int y, int[] array)
    {
        var temp = array[x];
        array[x] = array[y];
        array[y] = temp;
    }

    public static bool BubbleSortStep(int[] array)
    {
        if (array.Length < 2)
            return true;

        bool noSwaps = true;

        for (int i = 1; i < array.Length; i++)
        {
            if (array[i - 1] > array[i])
            {
                Swap(i, i - 1, array);
                noSwaps = false;
            }
        }

        return noSwaps;
    }
}

File: algos3-python/src/Task1.py
def SelectionSortStep(array, i):
    if i >= len(array) - 1:
        return
    min_index = i
    for j in range(i + 1, len(array)):
        if array[min_index] > array[j]:
            min_index = j
    array[i], array[min_index] = array[min_index], array[i]


def BubbleSortStep(array):
    if len(array) < 2:
        return True
    no_swaps = True
    for i in range(1, len(array)):
        if array[i - 1] > array[i]:
            no_swaps = False
            array[i - 1], array[i] = array[i], array[i - 1]
    return no_swaps
